src/biokbase/narrative/jobs/jobcomm.py

```diff
@@ -28,58 +30,54 @@ class JobRequest(object):
     """
     A small wrapper for job comm channel request data.
     This generally comes in the form of a packet from the kernel Comm object.
-    It's expected to be a dict of the format:
+    It is expected to be a dict of the format:
     {
         content: {
             'comm_id': <some string>,
             'data': {
-                'request_type': effectively, the function requested.
+                'request_type': job function requested
                 'job_id': optional
                 'job_id_list': optional
+                'batch_id': optional
             }
         }
     }
 
-    This little wrapper fills 2 roles:
+    This little wrapper fills some roles:
     1. It validates that the packet is correct and has some expected values.
-    2. If there's a job_id field, it makes sure that it's real (by asking the
-       JobManager - avoids a bunch of duplicate validation code)
-
-    Each JobRequest has at most one of a job_id or job_id_list. If the job comm
-    channel request data is received with a job_id_list, it may be split up
-    into multiple JobRequests. Likewise, if the job comm channel request data
-    is received with a job_id, a JobRequest may be created with a job_id_list
-    containing that job_id
+    2. It provides any job ID information with guardrails
 
     Provides the following attributes:
     raw_request     dict    the unedited request received by the job comm
     msg_id          str     unique message id
     request_type    str     the function to perform. This isn't
-        strictly controlled here, but by JobComm._handle_comm_message.
+                            strictly controlled here, but by
+                            JobComm._handle_comm_message.
     rq_data         dict    the actual data of the request. Contains the request
-        type and other parameters specific to the function to be performed
+                            type and other parameters specific to the function
+                            to be performed
 
-    Optional:
+    The IDs of the job(s) to perform the function on (optional):
     job_id          str
     job_id_list     list(str)
     batch_id        str
-
-    The IDs of the job(s) to perform the function on.
-
     """
 
+    INPUT_TYPES = [PARAM["JOB_ID"], PARAM["JOB_ID_LIST"], PARAM["BATCH_ID"]]
+
     def __init__(self, rq: dict):
-        self.raw_request = copy.deepcopy(rq)
+        rq = copy.deepcopy(rq)
+        self.raw_request = rq
         self.msg_id = rq.get("msg_id")  # might be useful later?
         self.rq_data = rq.get("content", {}).get("data")
-        if self.rq_data is None:
+        if not self.rq_data:
             raise JobRequestException(INVALID_REQUEST_ERR)
         self.request_type = self.rq_data.get("request_type")
-        if self.request_type is None:
+        if not self.request_type:
```

Comment:
I think the falsy values, `{}` or `""`, should be just as invalid and should be handled instead of just let through